sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -155,9 +155,15 @@ def rank_error(msg):
               or str(response.currentState) == 'JOB_STATE_UPDATED'
               or str(response.currentState) == 'JOB_STATE_DRAINED'):
             break
-          # The job has failed; ensure we see any final error messages.
-          sleep_secs = 1.0      # poll faster during the final countdown
-          final_countdown_timer_secs -= sleep_secs
+
+          # Check that job is in a post-preparation state before starting the
+          # final countdown.
+          if (str(response.currentState) != 'JOB_STATE_PENDING'
+              and str(response.currentState) != 'JOB_STATE_QUEUED'):
```

Comment:
I can't find any mention of `JOB_STATE_QUEUED` in https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs#jobstate

Reply:
I did find it documented here though: https://godoc.org/google.golang.org/api/dataflow/v1b3. Seems to be the same Dataflow API version, so it might just be a case of the documentation at https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs#jobstate not being updated.

Reply:
JOB_STATE_QUEUED is a recently added state. Docs are likely out of date.